auslib/db.py

```diff
@@ -1101,6 +1105,13 @@ def update(self, where, what, changed_by, old_data_version, transaction=None, dr
         # We need to check each Scheduled Change that would be affected by this
         # to ensure the new row will be valid.
         for row in self.select(where=where, transaction=transaction):
+            # verify whether the scheduled change has already been completed or not. If completed,
+            # then cannot modify the scheduled change anymore.
+            completed_scheduled_change = self.select(where={"sc_id": row["sc_id"]},
+                                                     transaction=transaction, columns=["complete"])
```

Comment:
Are you sure you need to do this select? It looks like you should already have the complete row in "row".